packit_service/worker/handlers/bodhi.py

```diff
@@ -63,15 +78,92 @@ def __init__(
+        koji_build_event = KojiBuildEvent.from_event_dict(self.data.event_dict)
+        return KojiBuildEventWrapper(
+            build_id=koji_build_event.build_id,
+            state=koji_build_event.state,
+            dist_git_branch=koji_build_event.git_ref,
+            nvr=koji_build_event.nvr,
+        )
+
+    def _build_from_koji_api(self) -> KojiBuildEventWrapper:
+        dist_git_branch = self.project.get_pr(self.data.pr_id).target_branch
+        nvr = DistGit.get_latest_build_in_tag(
+            downstream_package_name=self.project.repo, dist_git_branch=dist_git_branch
+        )
+
+        session = ClientSession(baseurl=KOJI_BASEURL)
+        build = session.getBuild(buildInfo=nvr)
+        return KojiBuildEventWrapper(
+            build_id=build["build_id"],
+            state=KojiBuildState.from_number(build["state"]),
+            dist_git_branch=dist_git_branch,
+            nvr=nvr,
+        )
+
+    def _build_koji_event_data(self) -> KojiBuildEventWrapper:
+        if self.data.event_type == PullRequestCommentPagureEvent.__name__:
+            return self._build_from_koji_api()
+
+        return self._build_from_event_dict()
 
     @property
-    def koji_build_event(self):
-        if not self._koji_build_event:
-            self._koji_build_event = KojiBuildEvent.from_event_dict(
-                self.data.event_dict
+    def koji_build_event_data(self) -> Optional[KojiBuildEventWrapper]:
+        if self._koji_build_event_data is None:
+            self._koji_build_event_data = self._build_koji_event_data()
+
+        return self._koji_build_event_data
+
+    @property
+    def packit_api(self) -> Optional[PackitAPI]:
+        if self._packit_api is None:
+            self._packit_api = PackitAPI(
+                self.service_config,
+                self.job_config,
+                downstream_local_project=self.local_project,
+            )
+
+        return self._packit_api
+
+    @property
+    def local_project(self) -> LocalProject:
+        if self._local_project is None:
+            self._local_project = LocalProject(
+                git_project=self.project,
+                working_dir=self.service_config.command_handler_work_dir,
+                cache=RepositoryCache(
+                    cache_path=self.service_config.repository_cache,
+                    add_new=self.service_config.add_repositories_to_repository_cache,
+                )
+                if self.service_config.repository_cache
+                else None,
+            )
+
+        return self._local_project
+
+    def _write_access_to_dist_git_repo_is_not_needed_or_satisfied(self) -> bool:
+        if self.data.event_type != PullRequestCommentPagureEvent.__name__:
+            # not needed
+            return True
+
+        user = self.data.actor
+        logger.debug(
+            f"Bodhi update will be re-triggered via dist-git PR comment by user {user}."
+        )
+
+        if not self.project.has_write_access(user=user):
```

Comment:
we just discussed with Franta and Tomas that here we should check both whether the user is a packager and has write access to the repo, so here we need to call the `is_packager`(https://github.com/packit/packit-service/blob/85b02fc86ea7a3591cbce3306a807eff261da616/packit_service/worker/handlers/distgit.py#L443)